dev/tools/controllerbuilder/template/controller/controller.go

```diff
@@ -254,17 +254,22 @@ func (a *Adapter) Export(ctx context.Context) (*unstructured.Unstructured, error
 	if mapCtx.Err() != nil {
 		return nil, mapCtx.Err()
 	}
-	// TODO(user): Update other resource reference 
+	// TODO(user): Update other resource references 
 	parent, err := a.id.Parent()
 	if err != nil {
 		return nil, err
 	}
-	obj.Spec.ProjectRef = &refs.ProjectRef{Name: parent.ProjectID}
+	obj.Spec.ProjectRef = &refs.ProjectRef{External: parent.String()}
 	obj.Spec.Location = parent.Location
 	uObj, err := runtime.DefaultUnstructuredConverter.ToUnstructured(obj)
 	if err != nil {
 		return nil, err
 	}
+
+	u.SetName(a.actual.Id)
```

Comment:
Maybe add a TODO comment about "Remove underscore and other chars that not allowed in the K8s naming convention."

Reply:
let's add an issue to track this instead 🙏🏼